aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -1027,11 +1080,55 @@ module aptos_framework::delegation_pool {
         });
     }
 
+    /// Allows an owner to update the commission percentage for the operator of the underlying stake pool.
+    public entry fun update_commission_percentage(
+        owner: &signer,
+        new_commission_percentage: u64
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        assert!(features::commission_change_delegation_pool_enabled(), error::invalid_state(
+            ECOMMISSION_RATE_CHANGE_NOT_SUPPORTED
+        ));
+        assert!(new_commission_percentage <= MAX_FEE, error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE));
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        assert!(
+            operator_commission_percentage(pool_address) + MAX_COMMISSION_INCREASE >= new_commission_percentage,
+            error::invalid_argument(ETOO_LARGE_COMMISSION_INCREASE)
+        );
+        assert!(
+            stake::get_remaining_lockup_secs(pool_address) >= min_remaining_secs_for_commission_change(),
+            error::invalid_state(ETOO_LATE_COMMISSION_CHANGE)
+        );
+
+        // synchronize delegation and stake pools before any user operation
+        // ensure the operator is paid its uncommitted commission rewards
+        // with the old commission percentage
+        synchronize_delegation_pool(pool_address);
+
+        let delegation_pool = borrow_global<DelegationPool>(pool_address);
+        let pool_signer = account::create_signer_with_capability(&delegation_pool.stake_pool_signer_cap);
+        if (exists<NextCommissionPercentage>(pool_address)) {
+            let commission_percentage = borrow_global_mut<NextCommissionPercentage>(pool_address);
+            commission_percentage.commission_percentage_next_lockup_cycle = new_commission_percentage;
+            commission_percentage.effective_after_secs = stake::get_lockup_secs(pool_address);
+        } else {
+            move_to(&pool_signer, NextCommissionPercentage {
+                commission_percentage_next_lockup_cycle: new_commission_percentage,
+                effective_after_secs: stake::get_lockup_secs(pool_address),
+            });
+        };
+
+        event::emit(CommissionPercentageChange {
+            pool_address,
+            operator: stake::get_operator(pool_address),
```

Comment:
would log the pool owner instead of the operator, as that's the account initiating the commission change

Reply:
@alexfilip2 , instead of `operator` or in addition to `opreator`?

Reply:
_instead of operator_, logging the current operator at the time of the commission update doesn't give us any unique information. Operator/beneficiary can change too by the time this new commission value becomes effective and that would be tracked from other events.

Reply:
Replaced `operator` with `owner`.